### DISCUSSION TITLE
Integration with Stable baselines?

### DISCUSSION THREAD

**Discussion by Rajivrocks-Ltd** [trackmania-rl/tmrl#101] [Q&A]
Hello, I was wondering, to make this work with stable baselines you need an OpenAI Gym environment. Does this mean I would need to write my own gym environment? Because now I get errors when I try to pass the "cfg_obj.ENV_CLS" I get errors which point me to that it's not an OpenAI gym enviroment.

**Comment by yannbouteiller** [comment]
Hi,

cfg_obj.ENV_CLS is a partially instantiated class, you need to instantiate it to get a Gymnasium environment.

Or you can use tmrl.get_environment()

I doubt that stable baselines implementations behave well with real-time environments though. I expect they step the environment once in a while and train in between, which doesn't work in real-time robotics. I would rather suggest to copy their algorithm implementations and adapt them to the tmrl `TrainingAgent`.

**Reply by Rajivrocks-Ltd** [reply]
Ah okay, thanks for your input!
